### PR TITLE
Release details feedback

### PR DESCRIPTION
**Pull request by jabrah**
Related: https://github.com/eclipse-pass/main/issues/420

The exact script is intended to inform GH automation as opposed to being run as-is.

Especially looking for feedback around when git commits need to be made and the `mvn versions` calls. Please ensure general correctness as well. 

Mostly opening this for feedback, once done, I'd be fine with either merging or closing without merge

### CODE REVIEW

**Review by jrmartino — APPROVED** [review]
this looks good to me. actually performing the releases according to this procedure  (even if automated) may require some additional changes or comments

**Review by markpatton — APPROVED** [review]
I haven't tried to go through it, but it looks pretty complete to me.

File: docs/dev/release.md [hunk 1]
```diff
@@ -124,6 +124,93 @@ At this point, we should have deployed the release to Sonatype (and eventually t
...
+* Maven deploy new snapshot
+* Push snapshot Docker image to GHCR
+
+This follows the same general release procedure as `main` with a few extra steps to account for submodules and Docker images
+
+### `pass-support`
+
+See procedure for `pass-core`, but without Docker images
+
+``` sh
```
Comment (aforward) [inline docs/dev/release.md:155]: Thoughts on moving this into a `RELEASE=0.2.0 NEXT=0.3.0-SNAPSHOT ./tools/release/xxxx` script to make the release a _one-liner_?

Comment (jabrah) [reply docs/dev/release.md:155]: Sure, I'd be for moving this over there. I had initially only wanted feedback about the completeness and correctness of the procedure in order to then translate it into a GH workflow, but I can see having a more full bash script like this available in `./tools` could be helpful as a fall back to the GH automation, especially if a GH automation isn't complete before the next release.

If we want this to serve as a one-liner for release it'll need some tweaks, but I think it's relatively close in its current state

**Review by jrmartino — APPROVED** [review]

**Pull request merged**